Thread-safe creation of temp dir

lsarra-ax opened the pull request:
Handle the exception triggered when the temp folders already exist in a multi-threaded setting 

See 
https://stackoverflow.com/questions/42544885/error-when-mkdir-in-multi-threads-in-python

lsarra-ax committed:
Thread-safe creation of temp dir

Handle the exception triggered when the temp folders already exist in a multi-threaded setting 

See 
https://stackoverflow.com/questions/42544885/error-when-mkdir-in-multi-threads-in-python

tylerflex commented:
Thanks @lsarra-ax for contributing this. @lei-flex @momchil-flex could you take a look and see what you think?

momchil-flex commented on tidy3d/web/cli/app.py:
```diff
@@ -16,8 +16,13 @@
+    os.makedirs(TIDY3D_DIR)
+except OSError as e:
+    if e.errno != errno.EEXIST:
+        raise  # Reraise if failed for reasons other than existing already
```
Thanks. This seems equivalent to `os.makedirs(TIDY3D_DIR, exist_ok=True)`. According to the link you shared, this version is only needed for python < 3.2, but we require >= 3.9, so let's just use the simple version. Could you update?

lsarra-ax replied:
Thanks for your reply! Sure, this should work as well

lsarra-ax committed:
prevent race condition - use python>3.9 solution

Just using os.makedirs with exist_ok=True to prevent errors when temp dir has already been created

lsarra-ax committed:
(remove unused dependence)

momchil-flex reviewed (approved):
Thanks. Merging into my branch so I can squash the commits and add a changelog item before merging in the main code (pre/2.8, not develop, as we will be releasing the 2.8.0rc1 version today).
